### ISSUE TITLE
Some DLCs cannot be used for synced list

### ISSUE THREAD

**Issue by devPeete** [Inklay/ETS2-ATS-Sync-Helper#9]
**Describe the bug**
Choosign DLCs "Viva la France!" and /or "Italia" will provocate the message of resetting the stats ingame in ETS2, hence new routes will be created from the game, the synced list will be overwritten. 

**To Reproduce**
1. Open ETS2 and the Sync Helper
2. Choose Steam and Settings folder
3. sace format Text shall appear
4. Select DLCS "Viva la France" and / or "Italia"
5. click Sync, load Save Game in ETS2 
6. You will see, the game resettet the maps due to changes in the save game
7. Synced list was overwritten

Reproduced on 2 indepentend systems

**Setup information**
- The game you're playing on (either ATS or ETS2): ETS2
- The game version: 1.39
- The app version: latest (5.9.0
- The DLCs you have: All except Road to Black Sea, Iberia and Heart of Russia. The error effects only Italia and France
- Current date: 14.04.2021
![ets2_20210414_145350_00](https://user-images.githubusercontent.com/78875190/114713684-a8730280-9d31-11eb-95d9-e08e6c845179.png)

**Comment by Inklay** [comment]
I think this is due to the 1.40 ealry compatibility, I'll try to fix this this week. You won't have to download anything again since it will be a server side update. I'll keep you updated ;)

**Comment by Inklay** [comment]
Should be fixed since TruckerMP supports 1.40

**Comment by Mxrynd** [comment]
still doesn't work even after TruckerMP  update

**Comment by Inklay** [comment]
Mmh ok that's weird, I'm gonna check it out tomorrow, thanks for replying !

**Comment by Mxrynd** [comment]
any updates on this?

**Comment by Inklay** [comment]
I'm working on it, should be fixed this week, I'll keep you updated on any progress

**Comment by Mxrynd** [comment]
ok, we did little research with friend. ETS log says:
```
00:02:57.008 : <ERROR> [unit] File '<MEMFILE><UFS>/home/profiles/526564446576696C/save/1/game.sii':
00:02:57.008 : <ERROR> [unit] The unit '_nameless.197.67d5.fe70' of type 'job_offer_data' has dangling pointer (to 'cargo.car_ibe') in the attribute named 'cargo'.
00:02:57.008 : <ERROR> [unit] File '<MEMFILE><UFS>/home/profiles/526564446576696C/save/1/game.sii':
00:02:57.008 : <ERROR> [unit] The unit '_nameless.197.67d5.fe70' of type 'job_offer_data' has dangling pointer (to 'trailer.car_trans_o') in the attribute named 'trailer_variant'.
00:02:57.008 : <ERROR> [unit] File '<MEMFILE><UFS>/home/profiles/526564446576696C/save/1/game.sii':
00:02:57.008 : <ERROR> [unit] The unit '_nameless.197.67d5.fe70' of type 'job_offer_data' has dangling pointer (to 'trailer_def.car_ibe.car_trans_o') in the attribute named 'trailer_definition'.
00:02:57.008 : <ERROR> [unit] File '<MEMFILE><UFS>/home/profiles/526564446576696C/save/1/game.sii':
00:02:57.008 : <WARNING> [unit] The pointer to 'cargo.car_ibe' looks like dangling pointer.
00:02:57.008 : <ERROR> [unit] File '<MEMFILE><UFS>/home/profiles/526564446576696C/save/1/game.sii':
00:02:57.008 : <WARNING> [unit] The pointer to 'trailer.car_trans_o' looks like dangling pointer.
00:02:57.008 : <ERROR> [unit] File '<MEMFILE><UFS>/home/profiles/526564446576696C/save/1/game.sii':
00:02:57.008 : <WARNING> [unit] The pointer to 'trailer_def.car_ibe.car_trans_o' looks like dangling pointer.
00:02:57.008 : <ERROR> [unit] File '<MEMFILE><UFS>/home/profiles/526564446576696C/save/1/game.sii':
00:02:57.008 : <ERROR> [unit] The unit '_nameless.196.09b6.6520' of type 'job_offer_data' has dangling pointer (to 'cargo.car_ibe') in the attribute named 'cargo'.
00:02:57.008 : <ERROR> [unit] File '<MEMFILE><UFS>/home/profiles/526564446576696C/save/1/game.sii':
00:02:57.008 : <ERROR> [unit] The unit '_nameless.196.09b6.6520' of type 'job_offer_data' has dangling pointer (to 'trailer.car_trans_o') in the attribute named 'trailer_variant'.
00:02:57.008 : <ERROR> [unit] File '<MEMFILE><UFS>/home/profiles/526564446576696C/save/1/game.sii':
00:02:57.009 : <ERROR> [unit] The unit '_nameless.196.09b6.6520' of type 'job_offer_data' has dangling pointer (to 'trailer_def.car_ibe.car_trans_o') in the attribute named 'trailer_definition'.
```
We can see that ETS can not find some assets on client side during routes loading. cargo.car_ibe filename gives the cue that the problem is the lack of Iberial DLC files on client side. Looking for this filename in server repo (https://github.com/Inklay/Inklay.net/search?p=2&q=car_ibe) we can see that every italy/france list contains route, requiring this cargo and some baltic lists have it too (7th for example).
To sum up: I think that this problem affects only players that not own Iberia DLC and can be fixed removing routes to Iberia from lists. Unfortenutly I do not own Iberia DLC yet and can not proof that having it fixes problem.
P.S. full list of problematic assets: cargo.car_ibe, trailer.car_trans_o, trailer_def.car_ibe.car_trans_o

**Comment by Inklay** [comment]
I figured this out too, The solution to avoid that is to make the jobs using those cargo only avalaible to the people with the Iberia DLC, I'm generating new job lists wich fix this issue. Since I'm quite busy with my job I won't have any time to test it properly until wednesday, so if everything works, fix should be out wednesday. Thanks for helping me out with this issue !

**Comment by Inklay** [comment]
The fix should be done, I'm gonna test it tomorrow. Once again I'm sorry if I'm kinda slow to fix issues, I don't really have much time to work on it

**Comment by Inklay** [comment]
After more investigation, there was an other cargo  causing this issue (cargo.olive_tree), I'm generating new job list, the fix will be out today, once again thanks for your help and your patience

**Comment by Inklay** [comment]
The fix is deployed to the server, I'm closing the issue.
